src/utils.ts

```diff
@@ -242,3 +242,32 @@ export function writeFileSync(filePath: string, content: string) {
   fs.mkdirSync(path.dirname(filePath), { recursive: true })
   fs.writeFileSync(filePath, content)
 }
+
+/**
+ * Replaces TypeScript declaration file
+ * extensions (`.d.ts`, `.d.mts`, `.d.cts`)
+ * with their corresponding JavaScript variants (`.js`, `.mjs`, `.cjs`).
+ *
+ * @param dtsFilePath - The file path to be transformed.
+ * @returns The updated file path with the JavaScript extension.
+ *
+ * @internal
+ */
+export function replaceDtsWithJsExtensions(dtsFilePath: string) {
+  return dtsFilePath.replace(
+    /\.d\.(ts|mts|cts)$/,
+    (_, fileExtension: string) => {
+      switch (fileExtension) {
+        case 'ts':
+        case 'tsx':
```

Comment:
The regex above doesn't match `tsx`, so either we should remove `tsx` here, or add `tsx` in the regex above.

Reply:
sure!

Reply:
I did a test and `tsc` will compile `component.tsx` into `component.d.ts`, so it seems that `.d.tsx` isn't a thing. We should simply remove `case 'tsx'`.

Reply:
Sounds good!